src/__test__/index.test.ts

```diff
@@ -1,41 +1,272 @@
+      expect(entities).to.have.nested.property('[0].firstName', 'Adam')
+      expect(entities).to.have.nested.property('[0].lastName', 'Doe')
+      expect(entities).to.have.nested.property('[0].age', 21)
+      expect(entities).to.have.nested.property('[0].ID')
+    })
+    it('should able to check for an existing entity', async () => {
+      const has = await transaction!.has([existingPerson.ID])
+      expect(has).to.not.be.undefined
+      expect(has).to.have.property('exists', true)
+    })
+    it('should be able to find an existing entity', async () => {
+      const find = await transaction!.modelFindByID(existingPerson.ID)
+      expect(find).to.not.be.undefined
+      expect(find).to.haveOwnProperty('entity')
+      const entity: Person = JSON.parse(find.entity)
+      expect(entity).to.not.be.undefined
+      expect(entity).to.have.property('firstName', 'Adam')
+      expect(entity).to.have.property('lastName', 'Doe')
+      expect(entity).to.have.property('age', 21)
+      expect(entity).to.have.property('ID')
+      expect(entity).to.deep.equal(existingPerson)
+    })
+    it('should be able to save an existing entity', async () => {
+      existingPerson.age = 99
+      const saved = await transaction!.modelSave([existingPerson])
+      expect(saved).to.not.be.undefined
+      // @todo: Is this really what we want this response to look like? Empty object?
+      expect(saved).to.be.empty
+      const deleted = await transaction!.modelDelete([existingPerson.ID])
+      expect(deleted).to.not.be.undefined
+      // @todo: Is this really what we want this response to look like? Empty object?
+      expect(deleted).to.be.empty
+    })
+    it('should be able to close/end an transaction', async () => {
+      await transaction!.end()
+    })
+  })
+  describe('.listen', () => {
+    it('should stream responses.', async () => {
+      const create = await client.modelCreate(store.id, 'Person', [createPerson()])
+      const entities = create.entitiesList.map(entity => JSON.parse(entity))
+      const existingPerson: Person = entities.pop()
+      const events: number[] = []
+      let closer: Function = () => {
+        throw new Error('closer function not updated')
+      }
+      try {
+        closer = client.listen(store.id, 'Person', existingPerson.ID, reply => {
+          const entity = JSON.parse(reply.entity)
+          expect(entity).to.not.be.undefined
+          expect(entity).to.have.property('age')
+          expect(entity.age).to.be.greaterThan(21)
+          events.push(entity.age)
+        })
 
-    describe('registerSchema', () => {
-      it('should resolve', async () => {
-        expect(await client.registerSchema(store.id, 'Person', personSchema)).to.not.be.undefined
-      })
+        existingPerson.age = 30
+        await client.modelSave(store.id, 'Person', [existingPerson])
+        existingPerson.age = 40
+        await client.modelSave(store.id, 'Person', [existingPerson])
+        await sleep(1500)
```

Comment:
Just wait a bit for the saves to make it through.

Reply:
Awesome!